### ISSUE TITLE
Level Up Python Skills: PEP8 + Linters + IDE Setup

### ISSUE THREAD

**Issue by kingb** [jeff-vincent/flask-api#5]
One interesting advantage of Python is that there is an Official Style Guide for writing Python code called [PEP8](https://www.python.org/dev/peps/pep-0008/).

One really good way to level up your Python skills, which I see many many many developers skip over, even ones with 10+ years of experience, is getting good at enforcing PEP8 guidelines. If you get good at this, then your code will look and _feel_ much cleaner and more familiar to other developers.

The easiest, and in my opinion, the best way to do this is to make sure your IDE is set up to tell you when you are violating the code style your following. This way as you're typing, you get immediate feedback when you violate the coding style, which will train you to use the proper style as you're coding.

Here's an example of PyCharm's output using your `admin.py` file:
![image](https://user-images.githubusercontent.com/70168/67150130-81d12a00-f268-11e9-84cb-b3b55e9846d2.png)

Some IDEs like PyCharm can also detect logical errors by inferring types used and recognizing when you pass the wrong type to a function. For example, in the logic error shown in the screenshot above, PyCharm detected a problem. Looking closer, it's because your class methods are missing the `self` attribute. 

There are Python tools you can run on your code outside of an IDE, and often other IDEs support various of these tools. I recommend reading this to get a quick overview. Internally, we settled on `flask8` since everyone's IDE supported `flake8`. 
http://books.agiliq.com/projects/essential-python-tools/en/latest/linters.html

TL;DR: Properly set up your IDE to give you Style Guide and Logic feedback, and you'll learn how to follow the style guide through immediate feedback. Your code will look and feel cleaner as a result. It's a simple thing to do that's often ignored by developers.
